### PR TITLE
Test Code Rabbit

### PR DESCRIPTION
**Pull request by katiakub**
## Release Notes

- **New Features**
	- Added a new gem dependency: `lru_redux`.
	- Introduced a new benchmarking method in `ThemeRunner` for testing tokenization.
	- Added a new benchmarking suite for comparing expression parsing implementations.
  
- **Enhancements**
	- Improved parsing logic for `for` and `if` tags.
	- Enhanced error handling for incomplete expressions.
	- Updated `ForloopDrop` and `Cycle` classes to utilize keyword arguments for clarity.

- **Bug Fixes**
	- Fixed handling of negative float literals and emoji strings in expressions.

- **Tests**
	- Expanded test coverage for edge cases in tokenization and expression parsing.
	- Introduced new tests for the `cycle` tag functionality.

### CODE REVIEW

File: lib/liquid/expression.rb [hunk 1]
```diff
@@ -33,16 +35,130 @@ def self.parse(markup)
...
+        num_end_pos = markup.length if ss.eos?
+
+        if num_end_pos
+          # number ends with a number "123.123"
+          markup.byteslice(0, num_end_pos).to_f
+        else
+          # number ends with a dot "123."
+          markup.byteslice(0, first_dot_pos).to_f
+        end
+      end
```
[the remark this answers is not in the document]
Comment (katiakub) [reply lib/liquid/expression.rb:159]: @coderabbitai give me assistance

[the remark this answers is not in the document]
Comment (katiakub) [reply lib/liquid/expression.rb:159]: @coderabbitai give me assistance

[the remark this answers is not in the document]
Comment (katiakub) [reply lib/liquid/expression.rb:159]: @coderabbit prepare tests using rspec
